### PR TITLE
Procedures wrapping Google Natural Language API

### PR DESCRIPTION
**Pull request by mneedham**
Fixes #<Replace with the number of the issue, Mandatory>

One sentence summary of the change.

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  -
  -
  -

### CODE REVIEW

File: src/test/kotlin/apoc/nlp/gcp/GCPProceduresTest.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    YIELD response
+                    RETURN response
+                """.trimIndent(), mapOf("apiKey" to apiKey)) {
+            println(it.resultAsString())
+        }
+    }
+
+    @Test
+    fun `should extract entities as virtual graph`() {
+        neo4j.executeTransactionally("""CREATE (a:Article {id: 1234, body:${'$'}body})""", mapOf("body" to body))
```
Comment (conker84) [inline src/test/kotlin/apoc/nlp/gcp/GCPProceduresTest.kt:53]: same here

File: src/main/kotlin/apoc/nlp/gcp/GCPProcedures.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        )
+
+        val documentToGraph = DocumentToGraph(tx, GraphsConfig(graphConfig))
+        val graph = documentToGraph.create(response["entities"])
+
+        val mutableGraph = graph.graph.toMutableMap()
+
+        val nodes = (mutableGraph["nodes"] as Set<Node>).toMutableSet()
+        val relationships = (mutableGraph["relationships"] as Set<Relationship>).toMutableSet()
+        if(storeGraph) {
```
Comment (conker84) [inline src/main/kotlin/apoc/nlp/gcp/GCPProcedures.kt:64]: we can use a functional way to do this:
```suggestion
        val node = if(storeGraph) {
            mergeRelationships(sourceNode, nodes, classifyRelationshipType(config)).forEach { rel -> relationships.add(rel) }
            sourceNode
        } else {
            val virtualNode = VirtualNode(sourceNode, sourceNode.propertyKeys.toList())
            createRelationships(virtualNode, nodes, classifyRelationshipType(config)).forEach { rel -> relationships.add(rel) }
            virtualNode
        }
        nodes.add(node)
```

File: src/test/kotlin/apoc/nlp/gcp/GCPProceduresTest.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        @JvmStatic
+        fun beforeClass() {
+            TestUtil.registerProcedure(neo4j, GCPProcedures::class.java)
+            assumeTrue(apiKey != null)
+        }
+    }
+
+    @Test
+    fun `should extract entities`() {
+        neo4j.executeTransactionally("""CREATE (a:Article {body:${'$'}body})""", mapOf("body" to body))
```
[the remark this answers is not in the document]
Comment (mneedham) [reply src/test/kotlin/apoc/nlp/gcp/GCPProceduresTest.kt:34]: I thought it'll read the body variable from Kotlin if I do that, whereas I wanted it to read the Neo4j parameter?

**Comment by conker84** [comment]
LGTM thanks a lot @mneedham!

**Pull request merged**